vm/systemSmartContracts/governance.go

```diff
@@ -0,0 +1,872 @@
+		return vmcommon.UserError
+	}
+	minVeto, ok := big.NewInt(0).SetString(string(args.Arguments[2]), conversionBase)
+	if !ok || minVeto.Cmp(big.NewInt(0)) < 0 {
+		g.eei.AddReturnMessage("changeConfig third argument is incorrectly formatted")
+		return vmcommon.UserError
+	}
+	minPass, ok := big.NewInt(0).SetString(string(args.Arguments[3]), conversionBase)
+	if !ok || minPass.Cmp(big.NewInt(0)) < 0 {
+		g.eei.AddReturnMessage("changeConfig fourth argument is incorrectly formatted")
+		return vmcommon.UserError
+	}
+
+	scConfig, err := g.getConfig()
+	if err != nil {
+		g.eei.AddReturnMessage("changeConfig error " + err.Error())
+		return vmcommon.UserError
+	}
+
+	scConfig.NumNodes = numNodes.Int64()
+	scConfig.MinQuorum = int32(minQuorum.Int64())
+	scConfig.MinVetoThreshold = int32(minVeto.Int64())
+	scConfig.MinPassThreshold = int32(minPass.Int64())
+
+	marshalledData, err := g.marshalizer.Marshal(scConfig)
+	if err != nil {
+		g.eei.AddReturnMessage("changeConfig error " + err.Error())
+		return vmcommon.UserError
+	}
+	g.eei.SetStorage([]byte(governanceConfigKey), marshalledData)
+
+	return vmcommon.Ok
+}
+
+func (g *governanceContract) getConfig() (*GovernanceConfig, error) {
+	marshalledData := g.eei.GetStorage([]byte(governanceConfigKey))
+	scConfig := &GovernanceConfig{}
+	err := g.marshalizer.Unmarshal(scConfig, marshalledData)
+	if err != nil {
+		return nil, err
+	}
+
+	return scConfig, nil
+}
+
+func (g *governanceContract) whiteListProposal(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	currentNonce := g.eei.BlockChainHook().CurrentNonce()
+	if currentNonce == 0 {
+		return g.whiteListAtGenesis(args)
+	}
+	if args.CallValue.Cmp(g.baseProposalCost) != 0 {
+		g.eei.AddReturnMessage("invalid callValue, needs exactly " + g.baseProposalCost.String())
+		return vmcommon.OutOfFunds
+	}
+	err := g.eei.UseGas(g.gasCost.MetaChainSystemSCsCost.Proposal)
+	if err != nil {
+		g.eei.AddReturnMessage("not enough gas")
+		return vmcommon.OutOfGas
+	}
+	if len(args.Arguments) != 3 {
+		g.eei.AddReturnMessage("invalid number of arguments")
+		return vmcommon.FunctionWrongSignature
+	}
+	if g.proposalExists(args.Arguments[0]) {
+		g.eei.AddReturnMessage("cannot re-propose existing proposal")
+		return vmcommon.UserError
+	}
+	if g.isWhiteListed(args.CallerAddr) {
+		g.eei.AddReturnMessage("address is already whitelisted")
+		return vmcommon.UserError
+	}
+
+	startVoteNonce, endVoteNonce, err := g.startEndNonceFromArguments(args.Arguments[1], args.Arguments[2])
+	if err != nil {
+		g.eei.AddReturnMessage("invalid start/end vote nonce " + err.Error())
+		return vmcommon.UserError
+	}
+
+	key := append([]byte(proposalPrefix), args.CallerAddr...)
+	whiteListAcc := &WhiteListProposal{
+		WhiteListAddress: args.CallerAddr,
+		ProposalStatus:   key,
+	}
+
+	key = append([]byte(whiteListPrefix), args.CallerAddr...)
+	generalProposal := &GeneralProposal{
+		IssuerAddress:  args.CallerAddr,
+		GitHubCommit:   args.Arguments[0],
```

Comment:
because you have to add an official page for others to see why a certain account want to whitelist itself. All the proposals are indexed in a nice page with the github commits. Where you can index the description and other data.